lib/patch_ruby.rb

```diff
@@ -3,7 +3,7 @@
 
 #The core API used to integrate with Patch's service
 
-The version of the OpenAPI document: 2
+The version of the OpenAPI document: 2.1.0
```

Comment:
I'm not sure we should update that. I don't know what it's for, but it looks like this was the major version before. @joustrich any insights?

Reply:
Also, it doesn't look this change was made in https://github.com/patch-technology/client-code-generation/pull/70/files (so would be overwritten by future changes). Was this a search and replace?

Reply:
Hey! Great question. I didn't manually change this in the SDKs so I assumed it was picking it up from somewhere in the client generation. Let me double check to see how it's being injected.

Reply:
OK, fine if that's the case, but that seems a little strange. I checked the `blame` and it looked like we had `v1` here for a very long time.

Reply:
Looking into it, it seems like it's picking up the version directly from [the swagger file here](https://github.com/patch-technology/patch/blob/a5113b3d50c5dc6dc501ce08f21c38047dd21053/openapi/v2/swagger.json#L19). I did update that value thinking we wanted to keep versions in sync, but let me know if we only want to maintain the major version!

Reply:
Can you check with @joustrich ?

Reply:
Yeah I believe that line in the swagger file should always be the API version, which will always be the major version API version will always be 1, 2, 3, 4, etc. We can release semantic version updates to SDKS for minor versions (which should be non-breaking changes), but breaking changes should probably be a major + API version update (v3 would be next) if there is no way to use field deprecation to make it a non-breaking change.


With this, it looks like we've kind of mixed API version and SDK version. It appears you're not attempting to release a new API version (otherwise we would have a whole new folder in `openapi/v3/swagger.json`) but are really overwriting v2, so this would still be returned with the `Patch-Version: 2` header but have  changed some of the API fields to point at the semantic version.

I think we need to revert some of the changes in `patch` to either keep this at V2 and make it backwards compatible or release a V3. @pcothenet @holtbp and @venalen what do you think? I am happy to jump on a huddle to discuss more.

Reply:
Thanks for the insight! We did opt to keep the current request/response as a v2 response, but the requests and responses are still backwards compatible. Despite the flag that we've labelled `vintage_year` as deprecated, we'll still be supporting that functionality and will still be including it in the response.

I can see how that would be confusing and perceived as a breaking change though. Perhaps we should remove "deprecated" if we'll still have support for it in v2?

Reply:
> Despite the flag that we've labelled vintage_year as deprecated, we'll still be supporting that functionality and will still be including it in the response.

I'm questioning whether we should be deprecating this at all btw. @bspellacy's [comment on an earlier PR](https://github.com/patch-technology/patch/pull/5418#issuecomment-1481684313) has caused me to marinate a bit more on the future design as it pertains to making requests for a single vintage year. 

I'm convinced the response should include `vintage_start_year` and `vintage_end_year` but for the request body, I've been mulling over the schema as we think about v3. Should we introduce a `vintage` nested object with 2 use cases: 1 for requesting a single vintage year and 1 for requesting a range?

```
{
  vintage: {
    year: 2021
  }
}


{
  vintage: {
    start_year: 2020,
    end_year: 2022
  }
}
```

Sorry, this is pretty late in the process to be introducing new concepts. No need to change direction now, we can revisit when cutting v3.

Reply:
Sounds like we are good to keep this in V2, so I think all we need to do is address Paul's comment and make sure we revert any API fields to still be 2 (that field in `openapi/v2/swagger.json`). If I remember correctly, swagger.json is auto generated, so there might be some other spot where we need to change 2.1.0 back to 2. The only version we want bumped is the SDK versions. In client-code-generation there are separate fields for API version and SDK version so that should be fine! Sorry it is a bit confusing with all of the different types of "versions" 😄

Reply:
@pcothenet Yeah, I'm thinking we pause on the API release for right now, rollout the rest of the changes (Dashboard & Admin) and then revisit later this week.

Reply:
@pcothenet yeah I think that is a good idea. I just went through the RFC and it is missing some of these lower-level details, so I will fix it up today and can re-socialize it